### ISSUE TITLE
Is it possible to outline a mesh and not other meshes in a scene?

### ISSUE THREAD

**Issue by sairam4123** [jocamar/Godot-Post-Process-Outlines#2]
Can you outline a particular mesh and not all meshes in a scene? I don't want to split up the meshes into different scene, if that's what you say?

**Comment by jocamar** [comment]
Yes, but it requires you to setup a separate render target. You'll need another viewport and setup a version of your scene there that mimics your main scene. All objects in this scene should have an unshaded flat color material, with the objects you want outlined having a different color. Then you use the outlines camera as the camera for that viewport with the use_bg_texture option set to true and solid color or transparent background. You can then access the texture from that viewport and overlay it on your "main" scene.

For an example of this technique you can see the demo on XRay3D here: https://github.com/GDQuest/godot-shaders/tree/master/godot

Another option is to just use a mesh outline instead of post-process outlines if it's just one object.

**Comment by sairam4123** [comment]
Well, I am implementing a building placement system, I would like to implement something like a outline, when a building is clicked, the selected building will be outlined until deselected. So the method you said is difficult because the outline is going to change and I plan to implement something like storeys, so I don't think the method you said will work.

**Comment by jocamar** [comment]
The outlined object changing isn't an issue with the method I said, you can just change what object has the different color material in the mirror scene to be the object that's currently selected.

But if you just want one object at a time to have the outline I'd go with the traditional mesh based outlines instead of a post-process effect like this, unless you find some sort of issue in the other solution. There are a lot of tutorials online about this, such as this one: https://www.youtube.com/watch?v=zJzcJ1Jzfdk
